src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -2,18 +2,97 @@
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int DEFAULT_INCREASE_CAPACITY = 2;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private int size;
+    private float threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    private static class Node<K, V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private int getHash(K key) {
+        return Math.abs(key == null ? 0 : (key.hashCode()));
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+        }
+        Node<K, V>[] tab = table;
+        int index = getHash(key) % tab.length;
+        int hash = getHash(key);
+        if (tab[index] == null) {
+            tab[index] = new Node<>(hash, key, value, null);
+            size++;
```

Comment:
you can `return;` and you won't need to write else after if

Reply:
Agree